tap_stripe/__init__.py

```diff
@@ -404,8 +419,16 @@ def sync_stream(stream_name):
     replication_key = metadata.get(stream_metadata, (), 'valid-replication-keys')[0]
     # Invoice Items bookmarks on `date`, but queries on `created`
     filter_key = 'created' if stream_name == 'invoice_items' else replication_key
-    stream_bookmark = singer.get_bookmark(Context.state, stream_name, replication_key) or \
-        int(utils.strptime_to_utc(Context.config["start_date"]).timestamp())
+
+    # Invoice was bookmarking on `date` but in latest API version, that field is deprecated and replication key changed to `created`
+    # kept `date` in bookmarking as it has to respect bookmark of active connection too
+    if stream_name == 'invoices':
+        stream_bookmark = singer.get_bookmark(Context.state, stream_name, 'date') or \
+            int(utils.strptime_to_utc(Context.config["start_date"]).timestamp())
```

Comment:
In the comment it says the bookmark has changed to Created date, whereas in the code you have added start_date? Is start_date similar to created_date? Please explain

Reply:
The replication key is changed to `created` but `date` was used previously so we are trying to get a bookmark from the state using the `date` key first and if not found in a state then use start_date as a bookmark.
https://github.com/singer-io/tap-stripe/blob/1223197217ab7c8a42c96d39474750bccf838dfd/tap_stripe/__init__.py#L425-L427

The previous code was also doing that like looking into state first and not found then use start_date.(Deleted line above)